staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -37,12 +37,34 @@ var ForeverTestTimeout = time.Second * 30
 // NeverStop may be passed to Until to make it never stop.
 var NeverStop <-chan struct{} = make(chan struct{})
 
-// StartUntil starts f in a new goroutine and calls done once f has finished.
-func StartUntil(stopCh <-chan struct{}, wg *sync.WaitGroup, f func(stopCh <-chan struct{})) {
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
+// Group is an interface to decouple code from sync.WaitGroup.
+type Group interface {
+	Add(delta int)
+	Done()
+}
+
+// StartWithChannelWithinGroup adds 1 to the group, starts f in a new goroutine and calls g.Done once f has finished.
+// stopCh is passed to f as an argument.
+func StartWithChannelWithinGroup(stopCh <-chan struct{}, g Group, f func(stopCh <-chan struct{})) {
```

Comment:
- [ ] s/stopCh/ch/  
  Per the discussion [here](https://github.com/kubernetes/kubernetes/pull/46094#discussion_r120114634), since this function makes no assumption about the meaning of the supplied channel, don't have its name imply anything about "stop[ping]." A valid caller could supply a function like this:
```go
func drainNoMoreThanFive(ch <-chan struct{}) {
  for i := 0; i != 5; i++ {
    if _, ok := <-ch; !ok {
      break
    }
  }
}

ch := ...
var wg sync.WaitGroup
wait.StartWithChannelWithinGroup(ch, wg, drainNoMoreThanFive)
```